packages/shared/lib/core/layer-2/utils/validateEthereumAddress.ts

```diff
@@ -0,0 +1,41 @@
+import { Keccak } from 'sha3'
+import { get } from 'svelte/store'
+
+import { localize } from '@core/i18n'
+import { networkHrp } from '@core/network'
+import { validateBech32Address } from '@core/utils'
+
+const HASH_SIZE = 256
+
+export function validateEthereumAddress(address: string): string {
+    if (!validateBech32Address(get(networkHrp), address)) {
```

Comment:
The reason for your confusion is our shitty error handling for address validation :D,[ I opened an issue to fix it](https://github.com/iotaledger/firefly/issues/5240), since it should go in a separate PR.